### PR TITLE
Moving yb cdc stream id to migration status record

### PR DESCRIPTION
**Pull request by priyanshi-yb**
Live migration pipeline- https://jenkins.dev.yugabyte.com/job/users/job/yb-voyager-testing/job/yb-voyager-testing-pipeline/1239/

### CODE REVIEW

File: migtests/scripts/live-migration-fallf-run-test.sh [hunk 1]
```diff
@@ -218,7 +218,7 @@ main() {
 	step "Clean up"
 
 	./cleanup-db
-	stream_id=$(cat "${EXPORT_DIR}/metainfo/yb_cdc_stream_id.txt")
+	stream_id=$(sqlite3 ${EXPORT_DIR}/metainfo/meta.db "select json_text from json_objects where key='migration_status';" | grep -oP '"YBCDCStreamID":"\K[^"]*')
```
Comment (amit-yb) [inline migtests/scripts/live-migration-fallf-run-test.sh:221]: Define a function to fetch value of any key from the msr:
```
get_value_from_msr() {
  jq_filter=$1
  val=`sqlite3 ${EXPORT_DIR}/metainfo/meta.db "select json_text from json_objects where key='migration_status';" | jq $jq_filter`
  return $val
}

echo get_value_from_msr('.["YBCDCStreamID"]')
```
Consider the above snippet as a guideline. Do test it before merging.

File: yb-voyager/src/dbzm/yb_cdc_client.go [hunk 2]
```diff
@@ -130,6 +126,12 @@ func (ybc *YugabyteDBCDCClient) DeleteStreamID() error {
 	if err != nil {
 		return fmt.Errorf("running command with args: %s, error: %s", args, err)
 	}
+	err = ybc.metaDB.UpdateMigrationStatusRecord(func(record *metadb.MigrationStatusRecord) {
+		record.YBCDCStreamID = ""
+	})
+	if err != nil {
+		return fmt.Errorf("failed to update migration status record: %w", err)
```
Comment (amit-yb) [inline yb-voyager/src/dbzm/yb_cdc_client.go:133]: A bit more specific:
`failed to clear CDC stream ID from the msr: %w`

File: yb-voyager/src/metadb/migrationStatus.go [hunk 3]
```diff
@@ -34,6 +34,7 @@ type MigrationStatusRecord struct {
 	FallBackSwitchProcessedByFBImporter        bool              `json:"FallBackSwitchProcessedByFFImporter"`
 	ExportSchemaDone                           bool              `json:"ExportSchemaDone"`
 	ExportDataDone                             bool              `json:"ExportDataDone"`
+	YBCDCStreamID                              string
```
Comment (amit-yb) [inline yb-voyager/src/metadb/migrationStatus.go:37]: json tag?

File: yb-voyager/src/dbzm/yb_cdc_client.go [hunk 4]
```diff
@@ -87,30 +89,24 @@ func (ybc *YugabyteDBCDCClient) GenerateAndStoreStreamID() (string, error) {
...
+		record.YBCDCStreamID = streamID
+	})
 	if err != nil {
-		return "", fmt.Errorf("creating file: %s, error: %s", streamIDFile, err)
-	}
-	defer file.Close()
-	_, err = file.WriteString(streamID)
-	if err != nil {
-		return "", fmt.Errorf("writing to file: %s, error: %s", streamIDFile, err)
+		return "", fmt.Errorf("failed to update migration status record: %w", err)
```
Comment (amit-yb) [inline yb-voyager/src/dbzm/yb_cdc_client.go:96]: More specific error message: `failed to save the newly created stream ID %q in the msr: %w`

**Review by amit-yb — APPROVED** [review]
LGTM with a few comments.

File: migtests/scripts/live-migration-fallf-run-test.sh (hunk 1 above)
Comment (priyanshi-yb) [reply migtests/scripts/live-migration-fallf-run-test.sh]: oh, thanks for this suggestion of `jq` tool.

**Pull request merged**